src/CommonLib/hostfxr_utility.cpp

```diff
@@ -297,18 +306,42 @@ HOSTFXR_UTILITY::SetHostFxrArguments(
     if (argv == NULL)
     {
         hr = E_OUTOFMEMORY;
-        goto Finished;
+        goto Failure;
+    }
+
+    argv[0] = SysAllocString(pstruExePath);
+    if (argv[0] == NULL)
+    {
+        goto Failure;
     }
+    dwArgsProcessed++;
 
-    argv[0] = SysAllocString(pstruExePath->QueryStr());
     argv[1] = SysAllocString(L"exec");
+    if (argv[1] == NULL)
+    {
+        goto Failure;
+    }
+    dwArgsProcessed++;
 
     for (INT i = 0; i < argc; i++)
     {
         argv[i + 2] = SysAllocString(pwzArgs[i]);
+        if (argv[1] == NULL)
+        {
+            goto Failure;
+        }
+        dwArgsProcessed++;
     }
 
     pConfig->SetHostFxrArguments(argc + 2, argv);
```

Comment:
SetHostFxrArguments may fail. Get the hresult value

Reply:
SetHostFxrArguments doesn't return an HRESULT and there is no way for it to fail right now.

Reply:
this SetHostFxrArguments is from Config class. I mistakenly read it with utility